day2/src/main.rs

```diff
@@ -0,0 +1,59 @@
+use itertools::Itertools;
+use std::time::Instant;
+
+const INPUT: &[u8] = include_bytes!("input.txt");
+
+fn parse(array: &[u8]) -> i32 {
+    if array.len() == 1 {
+        return (array[0] - b'0') as i32;
+    }
+    return (array[0] - b'0') as i32 * 10 + (array[1] - b'0') as i32;
+}
+
+fn report_is_safe(report: &Vec<i32>) -> bool {
+    let asc = report[1] > report[0];
+    report
+        .windows(2)
+        .find(|w| {
+            let diff = if asc { w[1] - w[0] } else { w[0] - w[1] };
+            diff < 1 || diff > 3
```

Comment:
Clippy will suggest this can be `!(1..=3).contains(&diff)` but I think yours is clearer.